plugins/org.eclipse.xtext/src/org/eclipse/xtext/workspace/FileWorkspaceConfig.xtend

```diff
@@ -93,4 +98,25 @@ class FileSourceFolder implements ISourceFolder {
 		'''�name� (�path�)'''
 	}
 	
+}
+
+@Data
+class SingleProjectWorkspaceConfig implements IWorkspaceConfig {
+	
+	IProjectConfig projectConfig
+	
+	override findProjectByName(String name) {
+		if (projectConfig.name == name)
+			return projectConfig
```

Comment:
Minor stylistic: I feel an explicit `return null` helps to understand the code better.